src/geom/Point.cpp

```diff
@@ -180,6 +180,19 @@ Point::getZ() const
     return coordinates.getAt<Coordinate>(0).z;
 }
 
+double
+Point::getM() const
+{
+    if(isEmpty()) {
+        throw util::UnsupportedOperationException("getM called on empty Point\n");
+    }
+    if(getCoordinatesRO()->hasM()) {
+        return coordinates.getAt<CoordinateXYZM>(0).m;
+    } else {
+        return std::numeric_limits<double>::quiet_NaN();
```

Comment:
I'm not sure if this is the correct approach, as I could not identify how `getZ()` returned nan for non-Z geometries. Suggestions welcome here.

Reply:
I'd be inclined to delegate this (and `getZ()`) to `CoordianteSequence::getOrdinate`.

That function looks like it may not handle XYM correctly -- it should dispatch on `CoordinateType`, not `hasZ` and `hasM`.

Reply:
Thanks, `coordinates.getOrdinate(0, CoordinateSequence::M)` is much simpler, but correct that XYM is not correctly handled. (XYZM is fine). I'll also refactor `GetZ` to use `getOrdinate`.